ddmark/validation.go

```diff
@@ -0,0 +1,176 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package ddmark
+
+import (
+	"fmt"
+	"reflect"
+
+	k8smarkers "sigs.k8s.io/controller-tools/pkg/markers"
+)
+
+// AllDefinitions contains all marker definitions for this package.
+var AllDefinitions []*k8smarkers.Definition
+var rulePrefix string = "ddmark:validation:"
+
+func init() {
+	addDefinition(Maximum(0), k8smarkers.DescribesField)
+	addDefinition(Minimum(0), k8smarkers.DescribesField)
+	addDefinition(Enum(nil), k8smarkers.DescribesField)
+	addDefinition(Required(true), k8smarkers.DescribesField)
+
+	addDefinition(ExclusiveFields(nil), k8smarkers.DescribesType)
+}
+
+// Maximum can applied to an int field and provides a (non-strict) maximum value for that field
+type Maximum int
+
+// Minimum can applied to an int field and provides a (non-strict) minimum value for that field
+type Minimum int
+
+// Enum can be applied to a string (or string-able) field and provides a restricted amount of possible values for that field
```

Comment:
Does this mean we could enum an int field?

Reply:
Yes, technically any kind is enum-able, following the interface conversion I recently added - haven't tested on structs though
I'll probably stick to string and int for now (which work fully) and will probably address full struct support later, as we don't need it rn